src/osyris/spatial/spatial.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/nvaytet/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
```

Comment:
Maybe we should think whether we want to allow extracting a sphere from a Datagroup as well, instead of just an entire dataset? E.g. calling `extract_sphere` on a Dataset would make a new Dataset, calling `extract_sphere` on a Datagroup would make a new Datagroup?

But maybe that would not be very useful? For instance, making maps would be broken for an extracted Datagroup, since it needs the positions from the `amr` group.

Reply:
Yeah I think we should stick to datasets, since there would be practically no use for extracting a sphere from a datagroup only (eg. if the user wants the azimuthal component of the velocities in the hydro group, we need the amr positions).